velox/exec/tests/HashJoinTest.cpp

```diff
@@ -5177,6 +5177,80 @@ TEST_F(HashJoinTest, dynamicFiltersAppliedToPreloadedSplits) {
       .run();
 }
 
+TEST_F(HashJoinTest, dynamicFiltersPushDownThroughAgg) {
+  const int32_t numRowsProbe = 300;
+  const int32_t numRowsBuild = 100;
+
+  // Create probe data
+  auto probeVectors = makeBatches(1, [&](int32_t /*unused*/) {
+    return makeRowVector({
+        makeFlatVector<int32_t>(
+            numRowsProbe, [&](auto row) { return row - 10; }),
+        makeFlatVector<int64_t>(numRowsProbe, folly::identity),
+    });
+  });
+  std::shared_ptr<TempFilePath> probeFile = TempFilePath::create();
+  writeToFile(probeFile->getPath(), probeVectors);
+
+  // Create build data
+  auto buildVectors = makeBatches(1, [&](int32_t /*unused*/) {
+    return makeRowVector(
+        {"u0", "u1"},
+        {
+            makeFlatVector<int32_t>(
+                numRowsBuild,
+                [&](auto row) { return 35 + 2 * (row + numRowsBuild / 5); }),
+            makeFlatVector<int64_t>(numRowsBuild, folly::identity),
+        });
+  });
+
+  createDuckDbTable("t", probeVectors);
+  createDuckDbTable("u", buildVectors);
+
+  auto probeType = ROW({"c0", "c1"}, {INTEGER(), BIGINT()});
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  auto buildSide =
+      PlanBuilder(planNodeIdGenerator).values(buildVectors).planNode();
+
+  // Inner join.
+  core::PlanNodeId probeScanId;
+  core::PlanNodeId joinId;
+  auto op = PlanBuilder(planNodeIdGenerator, pool_.get())
+                .tableScan(probeType)
+                .capturePlanNodeId(probeScanId)
+                .partialAggregation({"c0", "c1"}, {"sum(c1)"})
```

Comment:
Why not invalid query, using c1 both as grouping key and as input to aggregation is confusing. Is there any particular reason you are constructing a plan like this? If so, would you add a comment to explain?

Reply:
Has removed it